src/pair.ts

```diff
@@ -74,4 +90,18 @@ export abstract class PairXYeqK extends Pair {
 		return buckets[0].multipliedBy(outputAmount).div(
 			buckets[1].minus(outputAmount).multipliedBy(this.fee))
 	}
+
+	public snapshot(): PairXYeqKSnapshot {
+		return {
+			fee: this.fee.toFixed(),
+			bucketA: this.bucketA.toFixed(),
+			bucketB: this.bucketB.toFixed()
```

Comment:
during `snapshot()` always convert `BigNumber` to fixed string so that `JSON.stringify` and `JSON.parse` works correctly